### PR TITLE
Make sure to run tests on correct Go version (1.8.1)

### PR DESCRIPTION
**Pull request by zaksoup**
We want to make sure that the version of Golang in unit and integration test CI Builds is the version against which we compile the CPI when deploying a director. 

To that end we have added simple failure cases to the integration and unit tasks and set a version in the pipeline to match what is currently in the CI container.

This is our proposed solution, please discuss.

### CODE REVIEW

File: ci/tasks/run-int.yml [hunk 1]
```diff
@@ -11,17 +11,18 @@ inputs:
...
+  google_auto_network:           replace-me
+  google_target_pool:            replace-me
+  google_address_int:            replace-me
+  google_address_static_int:     replace-me
+  google_backend_service:        replace-me
+  google_region_backend_service: replace-me
+  google_service_account:        replace-me
   google_json_key_data: |
     replace-me
+  GO_VERSION: ""
```
Comment (johnsonj) [inline ci/tasks/run-int.yml]: `replace-me` for convention

File: ci/tasks/unit-tests.sh [hunk 2]
```diff
@@ -2,8 +2,16 @@
...
 set -e
 
+: ${GO_VERSION:?}
+
 export GOPATH=${PWD}/bosh-cpi-src
 export PATH=${GOPATH}/bin:$PATH
 
+current=$(go version)
+if [[ "$current" != *"$GO_VERSION"* ]]; then
+  echo "Go version is incorrect"
```
Comment (johnsonj) [inline ci/tasks/unit-tests.sh]: can you print out expected/found as part of the error

File: ci/pipeline.yml [hunk 3]
```diff
@@ -21,6 +21,9 @@ jobs:
 
       - task: unit-tests
         file: bosh-cpi-src/ci/tasks/unit-tests.yml
+        config:
+          params:
+            GO_VERSION: 1.8.1
 
   - name: build-candidate
```
Comment (johnsonj) [inline ci/pipeline.yml]: could this step verify the version of golang in the bosh release matches the container version? 

I like the spirit of this change but I believe this asserts the container runs a known version specified as an env variable. It doesn't verify the CPI is compiled with that version.

Comment (zaksoup) [inline ci/pipeline.yml]: Agreed, can do

File: ci/tasks/unit-tests.sh (hunk 2 above)
Comment (zaksoup) [inline ci/tasks/unit-tests.sh]: Yup

**Comment by zaksoup** [comment]
WRT to `replace-me`, our opinion has been that required values are left as empty strings in the task.yml and we check for presence using bash shorthand at the beginning of the task script:

```bash
: ${BOSH_ENVIRONMENT:?}
: ${BOSH_CLIENT:?}
: ${BOSH_CLIENT_SECRET:?}
```

We're not opposed to using the existing `replace-me` convention in this PR. That said, we feel that task.ymls should not be the place for documentation of required values, rather that strings like `replace-me` should exist in example pipelines and other explicit documentation. Our concern is that somebody not as familiar with concourse will see `replace-me` in a task.yml and think the task.yml is the correct place to modify/replace those values, rather than setting them in `config.params` on the pipeline task definitions.

**Comment by zaksoup** [comment]
@johnsonj we just pushed a commit that removes the `GO_VERSION` variable and infers the go version from the package spec for golang. Our previous comment about `replace-me` stands, but that's a PR for another day 😛 

Thanks!
zak && @cdutra

File: ci/tasks/run-int.sh [hunk 4]
```diff
@@ -77,6 +70,15 @@ export ZONE=${google_zone}
 export GOPATH=${PWD}/bosh-cpi-src
 export PATH=${GOPATH}/bin:$PATH
 
+release_go_version="$(cat "$GOPATH/packages/golang/spec" | \
```
Comment (johnsonj) [inline ci/tasks/run-int.sh]: awesome!!! 

now for nit: can you remove this duplication? either just run it as part of UTs or move it  to a function in `bosh-cpi-src/ci/tasks/utils.sh`

**Pull request merged**